plutus-ledger-api/src/PlutusLedgerApi/V3/Contexts.hs

```diff
@@ -0,0 +1,577 @@
+    vsep
+      [ "gaidTxId:" <+> pretty gaidTxId
+      , "gaidGovActionIx:" <+> pretty gaidGovActionIx
+      ]
+
+instance PlutusTx.Eq GovernanceActionId where
+  {-# INLINEABLE (==) #-}
+  GovernanceActionId a b == GovernanceActionId a' b' =
+    a PlutusTx.== a' PlutusTx.&& b PlutusTx.== b'
+
+data VotingProcedure = VotingProcedure
+  { vpVote   :: Vote
+  , vpAnchor :: Haskell.Maybe Anchor
+  }
+  deriving stock (Generic, Haskell.Show, Haskell.Eq)
+
+instance Pretty VotingProcedure where
+  pretty VotingProcedure{..} =
+    vsep
+      [ "vpVote:" <+> pretty vpVote
+      , "vpAnchor:" <+> pretty vpAnchor
+      ]
+
+instance PlutusTx.Eq VotingProcedure where
+  {-# INLINEABLE (==) #-}
+  VotingProcedure a b == VotingProcedure a' b' =
+    a PlutusTx.== a' PlutusTx.&& b PlutusTx.== b'
+
+data Committee = Committee
+  { committeeMembers :: Map ColdCommitteeHash Haskell.Integer
+  -- ^ Committee members with epoch number when each of them expires
+  , committeeQuorum  :: PlutusTx.Rational
+  -- ^ Quorum of the committee that is necessary for a successful vote
+  }
+  deriving stock (Generic, Haskell.Show, Haskell.Eq)
+
+instance Pretty Committee where
+  pretty Committee{..} =
+    vsep
+      [ "committeeMembers:" <+> pretty committeeMembers
+      , "committeeQuorum:" <+> pretty committeeQuorum
+      ]
+
+instance PlutusTx.Eq Committee where
+  {-# INLINEABLE (==) #-}
+  Committee a b == Committee a' b' =
+    a PlutusTx.== a' PlutusTx.&& b PlutusTx.== b'
+
+data Constitution = Constitution
+  { constitutionHash   :: PlutusTx.BuiltinByteString -- Balke2b_256 of some off-chain data
+  , constitutionScript :: Haskell.Maybe V2.ScriptHash
+  }
+  deriving stock (Generic, Haskell.Show, Haskell.Eq)
+
+instance Pretty Constitution where
+  pretty Constitution{..} =
+    vsep
+      [ "constitutionHash:" <+> pretty constitutionHash
+      , "constitutionScript:" <+> pretty constitutionScript
+      ]
+
+instance PlutusTx.Eq Constitution where
+  {-# INLINEABLE (==) #-}
+  Constitution a b == Constitution a' b' =
+    a PlutusTx.== a' PlutusTx.&& b PlutusTx.== b'
+
+data ProtocolVersion = ProtocolVersion
+  { pvMajor :: Haskell.Integer
+  , pvMinor :: Haskell.Integer
+  }
+  deriving stock (Generic, Haskell.Show, Haskell.Eq)
+
+instance Pretty ProtocolVersion where
+  pretty ProtocolVersion{..} =
+    vsep
+      [ "pvMajor:" <+> pretty pvMajor
+      , "pvMinor:" <+> pretty pvMinor
+      ]
+
+instance PlutusTx.Eq ProtocolVersion where
+  {-# INLINEABLE (==) #-}
+  ProtocolVersion a b == ProtocolVersion a' b' =
+    a PlutusTx.== a' PlutusTx.&& b PlutusTx.== b'
+
+data GovernanceAction
+  = -- | Proposed updates to PParams. This is a very tricky one, since PParamsUpdate is pretty
+    -- big and changes from one era to another. However, it could be very important for the
+    -- governance scripts to be able to see the values in the PParamsUpdate, but I am not
+    -- sure how to go forward about this, so for now I leave it as empty.
```

Comment:
This seems like a massive TODO...

Reply:
@lehins @WhatisRT Is it possible at all to make this contain only the delta, not the full params? Michael and I talked briefly about this and we suspect that including the full params here could be a serious problem due to the lack of efficient lookup operations in Plutus.